var/spack/repos/builtin/packages/unrar/package.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack import *
+
+
+class Unrar(MakefilePackage):
+    """RAR archive extraction utility"""
+
+    homepage = "http://www.rarlab.com"
+    url = "http://www.rarlab.com/rar/unrarsrc-5.8.1.tar.gz"
+
+    version(
+        "5.8.1",
+        sha256="035f1f436f0dc2aea09aec146b9cc3e47ca2442f2c62b4ad9374c7c9cc20e632",
+    )
+
+    def install(self, spec, prefix):
+        mkdir(prefix.bin)
+        install("unrar", prefix.bin)
```

Comment:
It looks like the makefile already has an `install` target. Why is this method necessary?

Reply:
The method is needed because the `install` method in the `makefile` hardcodes the install prefix...

Reply:
I think it's better to modify those hardcoded variables in an `edit` phase. For example, `LIBFLAGS` should be set to `self.compiler.pic_flag`, `CXX` may be fine but could be set to `spack_cxx`, and `DESTDIR` should be set to `prefix`.

Reply:
See https://spack.readthedocs.io/en/latest/build_systems/makefilepackage.html#edit-makefile for example docs.